pkg/controller/controller/group_controller.go

```diff
@@ -198,80 +202,96 @@ func GetAllGroups(crdclient *client.GroupCrdClient) []crd.Group {
-
-				podKey := pod.GetObjectMeta().GetNamespace() + ":" + pod.GetObjectMeta().GetName()
-				podDetails := group.Spec.PodsDetails
-
-				if podDetails == nil {
-					podDetails = map[string]*crd.PodDetails{}
-				}
-
-				existingPodDetails := podDetails[podKey]
-				if existingPodDetails != nil {
-					if payload.EventType == "create" {
-						// do nothing.
-					} else if payload.EventType == "delete" {
-						existingPodDetails.EndTime = pod.GetObjectMeta().GetCreationTimestamp()
-					}
-				} else {
-					if payload.EventType == "create" {
-						newPodDetails := crd.PodDetails{Name: pod.Name, StartTime: pod.GetCreationTimestamp()}
-						containers := []*crd.Container{}
-						for _, cont := range pod.Spec.Containers {
-							container := crd.Container{Name: cont.Name}
-							metrics := metrics.Metrics{}
-							if cont.Resources.Requests != nil {
-								metrics.CpuRequest = cont.Resources.Requests.Cpu()
-								metrics.MemoryRequest = cont.Resources.Requests.Memory()
-							}
-							if cont.Resources.Limits != nil {
-								metrics.CpuLimit = cont.Resources.Limits.Cpu()
-								metrics.MemoryLimit = cont.Resources.Limits.Memory()
-							}
-							container.Metrics = &metrics
-							containers = append(containers, &container)
-						}
-						newPodDetails.Containers = containers
-						podDetails[podKey] = &newPodDetails
-					} else if payload.EventType == "delete" {
-						// do nothing.
-					}
-				}
+				updatePodDetails(group, pod, payload)
 			}
 		}
 		log.Info("Completed updating User Created Groups for pod {} update.\n", pod.Name)
 	}
+}
 
-	// update all the groups
-	for _, group := range groups {
-		_, err := crdclient.UpdateGroup(&group)
+// nolint
+func updatePodDetails(group crd.Group, pod api_v1.Pod, payload eventprocessor.Payload) {
+	podKey := pod.GetObjectMeta().GetNamespace() + ":" + pod.GetObjectMeta().GetName()
+	podDetails := group.Spec.PodsDetails
 
-		if err != nil {
-			log.Error("There is an error while updating the crd for group = %s\n", group.Name)
-		} else {
-			log.Info("Updating the crd for group = %s is successful\n", group.Name)
+	if podDetails == nil {
+		podDetails = map[string]*crd.PodDetails{}
+	}
+
+	existingPodDetails := podDetails[podKey]
+	if existingPodDetails != nil {
+		if payload.EventType == Create {
+			// do nothing.
+		} else if payload.EventType == Delete {
+			existingPodDetails.EndTime = pod.GetObjectMeta().GetCreationTimestamp()
 		}
+	} else {
+		if payload.EventType == Create {
+			newPodDetails := crd.PodDetails{Name: pod.Name, StartTime: pod.GetCreationTimestamp()}
+			containers := []*crd.Container{}
+			for _, cont := range pod.Spec.Containers {
+				container := getPodMetrics(cont)
+				containers = append(containers, container)
+			}
+			newPodDetails.Containers = containers
+			podDetails[podKey] = &newPodDetails
+		} else if payload.EventType == Delete {
+			// do nothing.
+		}
+	}
+}
+
+func getPodMetrics(cont api_v1.Container) *crd.Container {
```

Comment:
`getContainerMetrics` or `getContainerWithMetrics` makes more sense for this function.